module/VuFind/src/VuFind/AjaxHandler/GetItemStatuses.php

```diff
@@ -474,64 +480,54 @@ protected function getItemStatusError($record, $msg = '')
     }
 
     /**
-     * Convert availability boolean or int to a string
+     * Get a message for availability status
      *
-     * @param mixed $available Boolean value or one of the ItemStatus::STATUS_*
-     *                         constants
+     * @param AvailabilityStatusInterface $availability Availability Status
      *
      * @return string
      */
-    protected function availabilityToString($available): string
+    protected function getAvailabilityMessage($availability): string
     {
-        switch ((int)$available) {
-            case ItemStatus::STATUS_UNAVAILABLE:
-                return 'false';
-            case ItemStatus::STATUS_UNCERTAIN:
-                return 'uncertain';
-            default:
-                return 'true';
+        $key = $availability->getStatusMessageKey();
```

Comment:
This feels a bit scary to me. Since `$availability` is not in control of the result of `renderAvailabilityMessage()`, it could be providing non-unique keys for different cases. Not a problem if `renderAvailabilityMessage()` just uses the status code, but if it incorporates anything else in the logic, it could lead to differences being lost.

Reply:
In the case that we don't know if the renderAvailabiltiyMessage returns different results we can return `null`. I added the key logic since in the current state the default status messages are also only rendered once.

Reply:
The current templates don't have any parameters, so they always return the exact same value. The problem for me isn't that there's some caching logic but that the key is created by another class that's not responsible for the rendering.